### PR TITLE
Update hyperlink for DefaultPromptFunctionText

### LINKED ISSUE #7795: `Get-Help Clear-Host -Online` gives broken link
Steps to reproduce
------------------

```powershell
get-help clear-host -Online
```

Expected behavior
-----------------

Should give proper links for help 

Actual behavior
---------------
Gives https://go.microsoft.com/fwlink/?LinkID=225747, which produces 404

Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.1.0
PSEdition                      Core
GitCommitId                    6.1.0
OS                             Linux 4.15.0-34-generic #37~16.04.1-Ubuntu SMP Tue Aug 28 10:44:06 UTC 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

Running on Elementary Loki

### PR DESCRIPTION
**Pull request by yecril71pl**
# PR Summary
The help hyperlink for `DefaultPromptFunctionText`, and hence `Prompt`, was invalid.  Fix #7795.

## PR Context

Invalid hyperlinks distract and annoy users and decrease productivity.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by KalleOlaviNiemitalo** [comment]
The same URL is here as well: <https://github.com/PowerShell/PowerShell/blob/9d05561d9c9788546050281a68456775d68fd7ef/test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv#L154>

**Comment by yecril71pl** [comment]
> The same URL is here as well:
> 
> https://github.com/PowerShell/PowerShell/blob/9d05561d9c9788546050281a68456775d68fd7ef/test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv#L154

This is a test data source.  The test `HelpSystem.OnlineHelp.Tests.ps1` is too weak—it does not verify that the help content makes sense.

**Comment by sdwheeler** [comment]
We should not remove this URL we need to fix it.

**Comment by sdwheeler** [comment]
@adityapatwardhan Do we want to fix the help for `prompt` or remove the link?

If we want to fix it, we need to create a new FWLink. The targe for that link should be point to `about_prompts` or we need to create a new `Microsoft.PowerShell.Core/prompt.md` file

**Comment by KalleOlaviNiemitalo** [comment]
> If we want to fix it, we need to create a new FWLink.

Would that make the existing URL redirect correctly (and thus fix PowerShell 6.2.7 as well), or would it require changing the `LinkID=225750` number?

**Comment by adityapatwardhan** [comment]
@sdwheeler I think we should fix the help content. Could you just update this link to point to the new content?

**Comment by sdwheeler** [comment]
The existing FWLink `LinkID=225750` does not exist anymore and cannot be recreated. We have to create a new one.

@adityapatwardhan Should we document the prompt function (new content) or point to about_Prompts? about_Prompts is more useful information and the plain function syntax that would be returned.

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 1]
```diff
@@ -4765,8 +4765,6 @@ internal static SessionStateAliasEntry[] BuiltInAliases
 
         internal const string DefaultPromptFunctionText = @"
 ""PS $($executionContext.SessionState.Path.CurrentLocation)$('>' * ($nestedPromptLevel + 1)) "";
```
Comment (adityapatwardhan) [inline src/System.Management.Automation/engine/InitialSessionState.cs:4767]:
```suggestion
""PS $($executionContext.SessionState.Path.CurrentLocation)$('>' * ($nestedPromptLevel + 1)) "";
# .Link
# https://go.microsoft.com/fwlink/?LinkId=2136843
```

File: test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv [hunk 2]
```diff
@@ -151,7 +151,6 @@ Out-Null,https://go.microsoft.com/fwlink/?linkid=2096792
 Out-Printer,https://go.microsoft.com/fwlink/?LinkID=2109553
 Out-String,https://go.microsoft.com/fwlink/?linkid=2097024
 Pop-Location,https://go.microsoft.com/fwlink/?linkid=2096907
```
Comment (adityapatwardhan) [inline test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv:153]:
```suggestion
Pop-Location,https://go.microsoft.com/fwlink/?linkid=2096907
Prompt,https://go.microsoft.com/fwlink/?LinkId=2136843
```

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 1 above)
Comment (adityapatwardhan) [reply src/System.Management.Automation/engine/InitialSessionState.cs:4767]: Added new link

File: test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv (hunk 2 above)
Comment (adityapatwardhan) [reply test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv:153]: Added new link

**Comment by adityapatwardhan** [comment]
@yecril71pl Please have a look at suggested changes.

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 1 above)
Comment (yecril71pl) [reply src/System.Management.Automation/engine/InitialSessionState.cs:4767]: This is not the correct link.  `HELP PROMPT` and `HELP PROMPT -O` should display the same content.

File: test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv (hunk 2 above)
Comment (sdwheeler) [reply test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv:153]: @yecril71pl We discussed this internally. The decision was made to point to the about topic.

Comment (yecril71pl) [reply test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv:153]: Please enlighten me regarding the prevailing argument.

Comment (vexx32) [reply test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv:153]: 1. There is a function named `prompt`
2. It makes sense for the help information for said function to point to information about prompts.

Is it so unclear?

Comment (yecril71pl) [reply test/powershell/engine/Help/assets/HelpURI/V2Cmdlets.csv:153]: It is unclear is why it should happen in the online version only.

**Comment by adityapatwardhan** [comment]
@sdwheeler Can you please review the comment help content.

**Comment by TravisEz13** [comment]
@sdwheeler  https://github.com/PowerShell/PowerShell/pull/13176#issuecomment-666577487

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 3]
```diff
@@ -4765,8 +4765,22 @@ internal static SessionStateAliasEntry[] BuiltInAliases
...
+# .OUTPUTS
+# The Prompt function must return an object.
+# As a best practice, return a string
+# or an object that is formatted as a string.
+# The maximum recommended length is 80 characters.
 # .Link
-# https://go.microsoft.com/fwlink/?LinkID=225750
+# https://go.microsoft.com/fwlink/?LinkId=2136843
+# .Link
+# about_Prompts
```
Comment (vexx32) [inline src/System.Management.Automation/engine/InitialSessionState.cs:4783]: Don't these help nodes require it to be all uppercase? Or am I misremembering. All the help docs I can recall  show them only with uppercase names (i.e., `.LINK`)

Comment (yecril71pl) [reply src/System.Management.Automation/engine/InitialSessionState.cs:4783]: This is the current declaration and it works.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@yecril71pl Thank you for your contribution!

**Pull request merged**